core/src/main/java/tachyon/client/BlockOutStream.java

```diff
@@ -108,21 +104,24 @@
 
   private synchronized void appendCurrentBuffer(byte[] buf, int offset, int length)
       throws IOException {
-    if (mAvailableBytes < length) {
-      long bytesRequested = mTachyonFS.requestSpace(mBlockId, length - mAvailableBytes);
-      if (bytesRequested + mAvailableBytes >= length) {
-        mAvailableBytes += bytesRequested;
-      } else {
-        mCanWrite = false;
-        throw new IOException(String.format("No enough space on local worker: fileId(%d)"
-            + " blockId(%d) requestSize(%d)", mFile.mFileId, mBlockId, length - mAvailableBytes));
+    initializeLocalFile();
```

Comment:
Yes, it would look good to me if initializeLocalFile throwing the IOException. As for BlockOutStream.close() method, I was thinking if we need check mLocalFile != null before calling cacheBlock in the close method. But in this case, it doesn't seem to be a problem as cancel() will be called if nothing is written.